isb_lib/vocabulary/vocab_adapter.py

```diff
@@ -11,13 +11,17 @@
 }
 """
 import logging
+from typing import Optional
+
 from term_store import TermRepository
 from term_store.db import Term
 
 VOCAB_CACHE: dict = {}
 
 
-def _read_descendants(term: Term, repository: TermRepository) -> dict:
+def _read_descendants(term: Term, repository: TermRepository, all_uris: set) -> Optional[dict]:
+    if term.uri in all_uris:
+        return None
```

Comment:
This works around the bug we saw where we were getting `Row` instead of `Term` which caused things to be visited multiple times.